src/main/java/org/folio/rest/impl/RequestsAPI.java

```diff
@@ -76,8 +76,12 @@ public void postRequestStorageRequests(
     Map<String, String> okapiHeaders,
     Handler<AsyncResult<Response>> asyncResultHandler,
     Context vertxContext) {
-
-    PgUtil.post(REQUEST_TABLE, entity, okapiHeaders, vertxContext,
+    Errors errors = RequestsApiUtil.validateRequest(entity);
+    if (!errors.getErrors().isEmpty()) {
+      asyncResultHandler.handle(succeededFuture(PostRequestStorageRequestsResponse
+        .respond422WithApplicationJson(errors)));
```

Comment:
Add return and move main logic outside of `else`

Reply:
Resolved